xarray/conventions/netcdftimeindex.py

```diff
@@ -106,9 +120,23 @@ def get_date_type(self):
     return type(self._data[0])
 
 
+def assert_all_same_netcdftime_datetimes(data):
+    from netcdftime._netcdftime import datetime
+
+    if not isinstance(data[0], datetime):
+        raise TypeError(
+            'NetCDFTimeIndex requires netcdftime._netcdftime.datetime'
```

Comment:
Unfortunately the public API name actually represents a ```DatetimeProlepticGregorian``` type, so for now to stick with public API imports, I've resorted to importing all six of the netcdftime datetime types.
```
In [1]: from netcdftime import datetime

In [2]: datetime(1, 1, 1)
Out[2]: netcdftime._netcdftime.DatetimeProlepticGregorian(1, 1, 1, 0, 0, 0, 0, -1, 1)
```